### PR TITLE
[Merged by Bors] - chore: forward port mathlib#18668, 18781, 18783, 18792, 18794

### PR DESCRIPTION
**Pull request by Parcly-Taxel**
* `set_theory.cardinal.basic`: forward-ports leanprover-community/mathlib#18781, which backports some ad-hoc changes in #3247 in a more principled way
* `ring_theory.int.basic`: these changes were already forward-ported in #3278, but we forgot the SHA.
* `linear_algebra.dimension`: forward-ports leanprover-community/mathlib#18792
* `linear_algebra.matrix.dot_product`: forward-ports leanprover-community/mathlib#18783
* `linear_algebra.finrank`: forward-ports leanprover-community/mathlib#18794 which is itself a backport of #3378, hence no changes to the file

---

* [`set_theory.cardinal.basic`@`e05ead7993520a432bec94ac504842d90707ad63`..`9bb28972724354ac0574e2b318be896ec252025f`](https://leanprover-community.github.io/mathlib-port-status/file/set_theory/cardinal/basic?range=e05ead7993520a432bec94ac504842d90707ad63..9bb28972724354ac0574e2b318be896ec252025f)
* [`ring_theory.int.basic`@`2196ab363eb097c008d4497125e0dde23fb36db2`..`e655e4ea5c6d02854696f97494997ba4c31be802`](https://leanprover-community.github.io/mathlib-port-status/file/ring_theory/int/basic?range=2196ab363eb097c008d4497125e0dde23fb36db2..e655e4ea5c6d02854696f97494997ba4c31be802)
* [`linear_algebra.dimension`@`45ce3929e3bf9a086a216feea3b1ab6c14bf0e67`..`b5b5dd5a47b5744260e2c9185013075ce9dadccd`](https://leanprover-community.github.io/mathlib-port-status/file/linear_algebra/dimension?range=45ce3929e3bf9a086a216feea3b1ab6c14bf0e67..b5b5dd5a47b5744260e2c9185013075ce9dadccd)
* [`linear_algebra.matrix.dot_product`@`738c19f572805cff525a93aa4ffbdf232df05aa8`..`46822d96c0c0a8e58a41a0cba1291620967575c5`](https://leanprover-community.github.io/mathlib-port-status/file/linear_algebra/matrix/dot_product?range=738c19f572805cff525a93aa4ffbdf232df05aa8..46822d96c0c0a8e58a41a0cba1291620967575c5)
* [`linear_algebra.finrank`@`8535b76e601f11868af3e612fbecb730998a5631`..`347636a7a80595d55bedf6e6fbd996a3c39da69a`](https://leanprover-community.github.io/mathlib-port-status/file/linear_algebra/finrank?range=8535b76e601f11868af3e612fbecb730998a5631..347636a7a80595d55bedf6e6fbd996a3c39da69a)

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/LinearAlgebra/Dimension.lean [hunk 1]
```diff
@@ -728,6 +728,11 @@ theorem linearIndependent_le_span {ι : Type _} (v : ι → M) (i : LinearIndepe
   exact le_top
 #align linear_independent_le_span linearIndependent_le_span
 
+/-- A version of `linearIndependent_le_span` for `Finset`. -/
+theorem linearIndependent_le_span_finset {ι : Type _} (v : ι → M) (i : LinearIndependent R v)
+    (w : Finset M) (s : span R (w : Set M) = ⊤) : (#ι) ≤ Fintype.card w := by
+  simpa only [Finset.coe_sort_coe, Fintype.card_coe] using linearIndependent_le_span v i w s
+
```
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/Dimension.lean:735]: Please don't forward-port things until the mathport output is ready (https://leanprover-community.github.io/mathlib-port-status/file/linear_algebra/dimension shows that it is not). Porting things manually leads to errors, and you've made one here.

Comment (Parcly-Taxel) [reply Mathlib/LinearAlgebra/Dimension.lean]: Done.

**Review by eric-wieser — APPROVED** [review]
I'll come back and merge this if CI passes.

**Comment by eric-wieser** [comment]
bors merge

**Pull request closed without merging**